test/e2e/common/util.go

```diff
@@ -184,3 +186,63 @@ func RestartNodes(c clientset.Interface, nodes []v1.Node) error {
 	}
 	return nil
 }
+
+func RecreateNodes(c clientset.Interface, nodes []v1.Node) error {
+	// Build mapping from zone to nodes in that zone.
+	nodeNamesByZone := make(map[string][]string)
+	for i := range nodes {
+		node := &nodes[i]
+		zone := framework.TestContext.CloudConfig.Zone
+		if z, ok := node.Labels[v1.LabelZoneFailureDomain]; ok {
+			zone = z
+		}
+		nodeNamesByZone[zone] = append(nodeNamesByZone[zone], node.Name)
+	}
+
+	// Find the sole managed instance group name
+	var instanceGroup string
+	if strings.Index(framework.TestContext.CloudConfig.NodeInstanceGroup, ",") >= 0 {
+		framework.Failf("Test does not support cluster setup with more than one managed instance group: %s", framework.TestContext.CloudConfig.NodeInstanceGroup)
+	} else {
+		instanceGroup = framework.TestContext.CloudConfig.NodeInstanceGroup
+	}
+
+	// Recreate the nodes.
+	for zone, nodeNames := range nodeNamesByZone {
+		args := []string{
+			"compute",
+			fmt.Sprintf("--project=%s", framework.TestContext.CloudConfig.ProjectID),
+			"instance-groups",
+			"managed",
+			"recreate-instances",
+			instanceGroup,
+		}
+
+		args = append(args, fmt.Sprintf("--instances=%s", strings.Join(nodeNames, ",")))
+		args = append(args, fmt.Sprintf("--zone=%s", zone))
+		framework.Logf("Recreating instance group %s.", instanceGroup)
+		stdout, stderr, err := framework.RunCmd("gcloud", args...)
+		if err != nil {
+			return fmt.Errorf("error restarting nodes: %s\nstdout: %s\nstderr: %s", err, stdout, stderr)
+		}
+	}
+
+	// Give a little time before polling the recreated nodes.
+	time.Sleep(pollInterval)
```

Comment:
Could you explain a bit more why we need this polling wait before the following bootID change wait?

Reply:
This appears to be unnecessary with the 10 minute node recreate timeout, which is longer than what I had tested with previously. Removed.

Originally, the test would fail without an additional wait before polling the bootID's, even with an extended timeout on the wait, which wasn't as long as 10 min.